File: src/CSharp/201-300/257.BinaryTreePaths.cs
using System.Collections.Generic;

namespace PeefyLeetCode.BinaryTreePaths
{
    public class TreeNode
    {
        public int val;
        public TreeNode left;
        public TreeNode right;
        public TreeNode(int x) { val = x; }
    }

public class Solution {
    public IList<string> BinaryTreePaths(TreeNode root) {
        var tostr = "->";
        List<string> l = new List<string>();
        if (root == null) {
            return l;
        }
        if (root.left == null && root.right == null){
            l.Add(root.val.ToString());
            return l;
        }
        var ll = BinaryTreePaths(root.left);
        var lr = BinaryTreePaths(root.right);
        for (int i = 0; i < ll.Count; i++)
        {
            l.Add(root.val.ToString() + tostr + ll[i]);
        }
        for (int i = 0; i < lr.Count; i++)
        {
            l.Add(root.val.ToString() + tostr + lr[i]);
        }
        return l;
    }
}

}

File: src/Java/201-300/257.BinaryTreePaths.java

import java.awt.List;
import java.util.ArrayList;
import java.util.LinkedList;
import java.util.Collections;
import java.util.Comparator;
import java.util.Queue;

public class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;

    TreeNode(int x) {
        val = x;
    }
}

class Solution {
    public List<String> binaryTreePaths(TreeNode root) {
        String to = "->";
        LinkedList<String> l = new LinkedList<>();
        if (root == null) {
            return l;
        }
        if (root.left == null && root.right == null){
            l.add(new Integer(root.val).toString());
            return l;
        }
        List<String> ll = binaryTreePaths(root.left);
        List<String> lr = binaryTreePaths(root.right);
        for (String s : ll){
            l.add(new Integer(root.val).toString() + to + s);
        }
        for (String s : lr){
            l.add(new Integer(root.val).toString() + to + s);
        }
        return l;
    }
}
